debezium-connector-sqlserver/src/main/java/io/debezium/connector/sqlserver/MaxLsnResult.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright Debezium Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.debezium.connector.sqlserver;
+
+/**
+ * Stores the result from querying the lsn_time_mapping for the highest LSNs.
+ */
+public class MaxLsnResult {
+
+    /**
+     * The highest lsn regardless of configuration. This should be utilized when querying the CDC tables as using an LSN that is no longer valid can lead to errors.
+     */
+    private final Lsn maxLsn;
+
+    /**
+     * The highest lsn belonging to a valid change transaction as determined by {@link SqlServerConnectorConfig#STREAMING_MAX_LSN_SELECT_STATEMENT} or default to the same as maxLsn.
+     */
+    private final Lsn maxTransactionalLsn;
+
+    public MaxLsnResult(Lsn maxLsn, Lsn maxTransactionalLsn) {
+        this.maxLsn = maxLsn;
+        this.maxTransactionalLsn = maxTransactionalLsn;
+    }
+
+    public Lsn getMaxLsn() {
+        return maxLsn;
+    }
+
+    public Lsn getMaxTransactionalLsn() {
```

Comment:
Do we actually need to expose both values to calling code? Wouldn't it suffice to expose only the lower of the two?

Reply:
We need both values because the stored proc later used to read the changes will fail if you give it an invalid lsn.
When query the actual tables, we need to always give it the real maximum lsn (and not just the highest one related to changes) otherwise if a server is dormant for a while, you'll run into errors.

I tried clarifying that through the comments, let me know what changes I can make to them as it's not an obvious thing and should be well documented in the code why.

Reply:
I think I got it overall, thanks for elaborating! One thing still unclear: when does an LSN become invalid exactly?

Reply:
From my understanding, LSNs are cleaned up and removed from the lsn_time_mapping table. It happens automatically, but I believe this stored procedure also could trigger it manually if you wanted to see an example 
https://docs.microsoft.com/en-us/sql/relational-databases/system-stored-procedures/sys-sp-cdc-cleanup-change-table-transact-sql?view=sql-server-ver15